gapis/api/vulkan/resources.go

```diff
@@ -773,7 +773,17 @@ func (cmd *VkCreateShaderModule) Replace(ctx context.Context, c *capture.Capture
 	cmd.Mutate(ctx, api.CmdNoID, state, nil)
 
 	shader := data.GetShader()
-	codeSlice := shadertools.AssembleSpirvText(shader.Source)
+	var codeSlice interface{}
+	var codeSize int
+	if shader.GetType() == api.ShaderType_Spirv {
+		assembledCode := shadertools.AssembleSpirvText(shader.Source)
+		codeSlice = assembledCode
+		codeSize = len(assembledCode) * 4
+	} else {
+		codeSlice = []byte(shader.Source)
```

Comment:
We should probably make sure this is a multiple of 4 bytes, or else we would be creating an invalid SPIR-V program / Vulkan trace. 

Nicer to tell the user they are doing something wrong earlier.

Reply:
Done